src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java

```diff
@@ -131,25 +131,19 @@ public static StarlarkOptionsParser newStarlarkOptionsParser(
   // require multiple rounds of parsing to fit starlark-defined options into native option format.
   @VisibleForTesting
   public void parse(ExtendedEventHandler eventHandler) throws OptionsParsingException {
-    ImmutableList.Builder<String> residue = new ImmutableList.Builder<>();
+    parseGivenArgs(eventHandler, nativeOptionsParser.getSkippedArgs());
+  }
+
+  @VisibleForTesting
+  public void parseGivenArgs(ExtendedEventHandler eventHandler, List<String> args)
+      throws OptionsParsingException {
     // Map of <option name (label), <unparsed option value, loaded option>>.
     Multimap<String, Pair<String, Target>> unparsedOptions = LinkedListMultimap.create();
 
-    // sort the old residue into starlark flags and legitimate residue
-    for (String arg : nativeOptionsParser.getPreDoubleDashResidue()) {
-      // TODO(bazel-team): support single dash options?
-      if (!arg.startsWith("--")) {
-        residue.add(arg);
-        continue;
-      }
-
+    for (String arg : args) {
       parseArg(arg, unparsedOptions, eventHandler);
     }
 
-    List<String> postDoubleDashResidue = nativeOptionsParser.getPostDoubleDashResidue();
-    residue.addAll(postDoubleDashResidue);
-    nativeOptionsParser.setResidue(residue.build(), postDoubleDashResidue);
```

Comment:
@fmeum the imported cl contains failures for tests related to the clean and info command due to the removal of these lines. The code for clean and info are looking at the residues https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/main/java/com/google/devtools/build/lib/runtime/commands/CleanCommand.java#L124 

Can we make sure that the behavior for clean and info do not change for starlark options after this change? 

To my surprise, the tests for these two commands were only available internally.

Reply:
I pushed a new commit that should restore the original behavior for the commands - please let me know whether the tests pass now. Along the way, I ended up deleting quite a bit of code that is no longer used.

Reply:
Thanks! @sgowroji is working with me on the import cl. We will update soon.